content/tutorials/aws-persistence.textile

```diff
@@ -26,15 +26,17 @@ meta_description: "How to persist Ably messages into AWS S3."
 meta_keywords: "Ably realtime, AWS Lambda Functions, integrations, AWS S3, database, persistence"
 ---
 
-Ably "webhooks":/general/webhooks enable users to configure custom rules that react to channel events. For example, triggering an action when a message is sent, or when a user joins the channel. You can use our integration with "AWS Lambda":https://aws.amazon.com/lambda/ to create these rules.
+Ably "webhooks":/general/webhooks enable users to configure custom rules that react to channel events such as messages being published or presence events from channels. These rules can notify HTTP endpoints, serverless functions or other services for each event.
 
-In this tutorial you will learn how to create a channel rule that intercepts messages sent to a channel and persists them to a file in "AWS Simple Storage Service (S3)":https://aws.amazon.com/s3/.
+One way to expand on these rules is by using our integration with "AWS Lambda":https://aws.amazon.com/lambda/ to create more advanced rules. For example, triggering an action that copies data to a third party storage system for persistence when a message is sent.
+
+In this tutorial you will learn how to create a channel rule that intercepts messages sent to a channel and persists them to a file in "AWS Simple Storage Service (S3)":https://aws.amazon.com/s3/. Note that S3 isn't optimised for high volumes of small messages, if you're sending many small messages you may want to try batching messages with "AWS Kinesis":https://ably.com/docs/general/firehose/kinesis-rule and setting Kinesis to "write to S3":https://docs.aws.amazon.com/firehose/latest/dev/create-destination.html#create-destination-s3 instead.
```

Comment:
> https://ably.com/docs/general/firehose/kinesis-rule

https://ably.com/docs not required. Also I would add a short sentence somewhere in this section pointing people at our Kinesis tutorial, that would be a very useful starting point for someone who wants to implement this.